src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -341,14 +346,10 @@ emit_simd_ins_for_binary_op (MonoCompile *cfg, MonoClass *klass, MonoMethodSigna
 			case SN_Divide:
 			case SN_op_Division: {
 				const char *class_name = m_class_get_name (klass);
-				if (!strcmp("Vector4", class_name) && fsig->params [1]->type == MONO_TYPE_R4) {
-					// Handles  Vector4 / scalar
-					return handle_mul_div_by_scalar (cfg, klass, MONO_TYPE_R4, args [1]->dreg, args [0]->dreg, OP_FDIV);	
-				}
-				if (strcmp ("Vector2", class_name) && strcmp ("Vector4", class_name) && strcmp ("Quaternion", class_name) && strcmp ("Plane", class_name)) {
-					if ((fsig->params [0]->type == MONO_TYPE_GENERICINST) && (fsig->params [1]->type != MONO_TYPE_GENERICINST))
+				if (strcmp ("Quaternion", class_name) && strcmp ("Plane", class_name)) {
```

Comment:
Why doesn't Vector2/4 require the logic in `handle_mul_div_by_scalar`?

Reply:
After this change, Vector2/4 go thru the same code path as other vectors (using the same logic for handling division by vector/scalar). After completing intrinsics support for Quaternion and Plane (https://github.com/dotnet/runtime/issues/88207) we will be able to remove the if statement completely probably.